src/compound.jl

```diff
@@ -0,0 +1,59 @@
+struct CompoundSpecies end
+Symbolics.option_to_metadata_type(::Val{:iscompound}) = CompoundSpecies
+
+struct CompoundComponents end
+Symbolics.option_to_metadata_type(::Val{:components}) = CompoundComponents
+
+
+macro compound(species_expr, arr_expr...)
+    # Ensure the species name is a valid expression
+    if !(species_expr isa Expr && species_expr.head == :call)
+        error("Invalid species name in @compound macro")
+    end
+
+    # Parse the species name to extract the species name and argument
+    species_name = species_expr.args[1]
+    species_arg = species_expr.args[2]
+
+    # Construct the expressions that define the species
+    species_expr = Expr(:macrocall, Symbol("@species"), LineNumberNode(0), Expr(:call, species_name, species_arg))
```

Comment:
Does this support interpolation of previously defined species? i.e.
```julia
@variables t
@species A(t) B(t)
@compound A2B3 2*($A) 3*($B)
```
to use the existing A and B symbolic variables instead of creating new ones? The reason to support this is that there may be extra metadata attached to predefined species we want to preserve and keep in the compound.
```

Reply:
Before defining a compound the constituent species must be explicitly declared. The `@compound` macro only creates a single new species which is the compound itself. So if I use a previously declared species with the macro, it is not created again and so the metadata is preserved. 

So in this case, any metadata previously attached to species `A` or `B` will be accessible even after it has been used with the `@compound` macro.